lib/ssh_scan/database/mongo_queue.rb

```diff
@@ -0,0 +1,49 @@
+require 'mongo'
+
+Mongo::Logger.logger.level = ::Logger::FATAL
+
+module SSHScan
+  module QueueDB
+    class MongoDb
+      attr_reader :queue_collection
+
+      def initialize(queue_client)
+        @queue_client = queue_client
+        @queue_collection = @queue_client[:ssh_scan_queue]
+      end
+
+      # Helps us create a SSHScan::DB::MongoDB object with a hash
+      def self.from_hash(opts)
+        name = opts["queue_db_name"]
+        server = opts["server"]
+        port = opts["port"]
+        socket = server + ":" + port.to_s
+
+        queue_client = Mongo::Client.new([socket], :database => name)
+        return SSHScan::QueueDB::MongoDb.new(queue_client)
+      end
+
+      # @param [String] socket
+      def push(socket)
+        @queue_collection.insert_one("socket" => socket, "status" => "pending")
```

Comment:
We are going to use the database to store scan results anyways, even if we switch job_queue later to something like redis, that is why I wanted to keep them de-coupled.